Task 3

gordoney opened the pull request:
Frontend part pull request: 
- https://github.com/gordoney/shop-react-redux-cloudfront/pull/1

Get Products Lambda: 
- https://8b7s2k085j.execute-api.eu-west-1.amazonaws.com/dev/products

Get Product By Id Lambda: 

- https://8b7s2k085j.execute-api.eu-west-1.amazonaws.com/dev/products/415f9828-f542-4bd5-a723-a328713473db
- https://8b7s2k085j.execute-api.eu-west-1.amazonaws.com/dev/products/3061fd99-3877-4443-bdb4-bc0ce4f8557e
- etc.



What was done:
- main parts are ready: getProductsById and getProductsList were created, frontend was updated
- additional tasks are ready: async/await is used, ES6 modules are used, Webpack is configured, openapi documentation are created, tests are created, lambda functions are separated, errors are handled.

gordoney committed:
Main part

gordoney committed:
'Additional-tasks'

gordoney committed:
individual-packaging

vishalviswanathan commented:
When I hit the endpoint for getProductList (https://8b7s2k085j.execute-api.eu-west-1.amazonaws.com/dev/products) I get a list of products with titles playstation. but when I checked the PR the productList.json file have different set of data. In FE side, am  seeing the data from productList.json. 

And for the getProductsById, when I hit url like this https://8b7s2k085j.execute-api.eu-west-1.amazonaws.com/dev/products/1 (as given in description) am not getting the data. But when I hit with this https://8b7s2k085j.execute-api.eu-west-1.amazonaws.com/dev/products/f13f0a9e-d101-4e29-88cf-ad88b5f7d8e2 am getting the product.

Can you please clarify the above two. I see there is a mismatch in data. Which product data is expected when hitting the endpoint and on FE?



For swagger, I see the the yaml configuration. But don't we need the endpoint for swagger docs? I find it missing.

For getProductById endpoint, when I give invalid productId it is not showing product not found error.



Rest all looks good to me.

gordoney commented:
Hello Vishal,

Sorry for mistakes) Some of them happened because I continued working on next tasks (with database) using the same serverless config)

1) I made local changes but forget to deployed them to AWS. I have done that, so you can see "playstation", etc products

2) That happened because I added connection to postgres instead of local file) I changed url to products to correct ones.

3) Fixed issue for 404 error for wrong productId

4) As I noticed on Q&A session, it looks like openapi.yml is enough
